### ISSUE TITLE
Stop custom sorting for theme.breakpoints

### ISSUE THREAD

**Issue by 7iomka** [mantinedev/mantine#4050]
### What package has an issue

@mantine/core

### Describe the bug

Hi.
I was just dumbfounded by the strange behavior with breakpoints.
I use tailwind and I have a common config with breakpoints.
```ts
import { em } from './lib';

export const breakpointsClean = {
  min: 0,
  xs: 360,
  xsm: 375,
  xsl: 414,
  sm: 576,
  smd: 640,
  md: 768,
  lg: 992,
  xl: 1200,
  lp: 1366,
  xxl: 1400,
};

export const breakpoints = Object.keys(breakpointsClean).reduce(
  (bps, key) => ({
    ...bps,
    [key]: em(breakpointsClean[key]),
  }),
  {} as Record<keyof typeof breakpointsClean, string>,
);


```

Used in theme config
```ts
import { breakpoints } from './config';

const themeConfig: MantineThemeOverride = {
  breakpoints,
}
```

And, I tried to use some Global styles like
```ts
import { containerMaxWidths } from './config';
import { Global } from '@mantine/core';

const AppGlobalStyles = () => (
  <Global
    styles={(theme) => ({
        // variables
        ':root': {
          // add variables for container max-width
          ...Object.keys(theme.breakpoints).reduce(
            (acc: any, curr: any) => ({
              ...acc,
              [theme.fn.largerThan(curr)]: {
                '--container-max-width': containerMaxWidths[curr] || theme.breakpoints[curr],
              },
            }),
            {},
          ),
          
      },
    })}
  />
);
```

Expected behaviour:
All breakpoints from `theme.breakpoints` have the same order as original `breakpoints`.

Actual behavior:
All breakpoints from `theme.breakpoints` **have different order**: the reserved keys (`xs`, `sm`, `md`, `lg`, `xl`) are displayed first, then the other keys are displayed in original order. 

So in my example I have unexpected broken code
<details>
  <summary>Generated styles</summary>
  

```css

@media (min-width: 22.5em) {
  :root {
    --container-max-width: 22.5rem;
  }
}

@media (min-width: 36em) {
  :root {
    --container-max-width: 33.75rem;
  }
}

@media (min-width: 48em) {
  :root {
    --container-max-width: 45rem;
  }
}

@media (min-width: 62em) {
  :root {
    --container-max-width: 60rem;
  }
}

@media (min-width: 75em) {
  :root {
    --container-max-width: 71.25rem;
  }
}

@media (min-width: 0em) {
  :root {
    --container-max-width: 20rem;
  }
}

@media (min-width: 23.4375em) {
  :root {
    --container-max-width: 23.4375rem;
  }
}

@media (min-width: 25.875em) {
  :root {
    --container-max-width: 25.875rem;
  }
}

@media (min-width: 40em) {
  :root {
    --container-max-width: 33.75rem;
  }
}

@media (min-width: 85.375em) {
  :root {
    --container-max-width: 71.25rem;
  }
}

@media (min-width: 87.5em) {
  :root {
    --container-max-width: 82.5rem;
  }
}

```

</details>

The only solution at the moment is to use the original `breakpoints` in this particular case, but we are using variables from the `theme.breakpoints` in multiple components and this will create confusion. Moreover, in many components this is done under the hood.

### What version of @mantine/hooks page do you have in package.json?

latest

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

### Possible fix

My suggestion is to remove this "smart sorting" from your code as it breaks the code.

**Comment by rtivital** [comment]
> All breakpoints from theme.breakpoints have the same order as original breakpoints

Objects keys in JavaScript do not have order. You should not rely on it while building your application.

**Comment by 7iomka** [comment]
> Objects keys

using the Object.keys() method to extract the keys of the theme.breakpoints object, the keys will be returned in the order they were added to the object. (String keys, in insertion order (ES2015 guarantees this and all browsers comply, node.js the same). This means that the loop will iterate over the breakpoints in the correct order.

Note, however, that the issue is not with the way I iterate, but with the way data for breakpoints is merged under the hood, and precisely because of the special nature of breakpoints, I propose to treat them in a special way instead of just merging them together with the default one, because this produce unexpected result when user provided breakpoints has more breakpoints keys than default one. And I'm sure there are many of them (especially those who use Tailwind).
I would be very grateful if you could approve my PR.

**Comment by rtivital** [comment]
Fixed in 6.0.8
